c10/cuda/CMakeLists.txt

```diff
@@ -12,71 +11,75 @@ configure_file(
     ${CMAKE_CURRENT_LIST_DIR}/impl/cuda_cmake_macros.h.in
     ${CMAKE_BINARY_DIR}/c10/cuda/impl/cuda_cmake_macros.h)
 
-# Note: if you want to add ANY dependency to the c10 library, make sure you
-# check with the core PyTorch developers as the dependency will be
-# transitively passed on to all libraries dependent on PyTorch.
+if(BUILD_LIBTORCHLESS)
+  find_library(C10_CUDA_LIB c10_cuda PATHS $ENV{LIBTORCH_LIB_PATH} NO_DEFAULT_PATH)
+else()
 
-# Note: if you add a new source file/header, you will need to update
-# torch/utils/hipify/cuda_to_hip_mappings.py for new files
-# and headers you add
-set(C10_CUDA_SRCS
-    CUDAAllocatorConfig.cpp
-    CUDACachingAllocator.cpp
-    CUDADeviceAssertionHost.cpp
-    CUDAException.cpp
-    CUDAFunctions.cpp
-    CUDAMallocAsyncAllocator.cpp
-    CUDAMiscFunctions.cpp
-    CUDAStream.cpp
-    impl/CUDAGuardImpl.cpp
-    impl/CUDATest.cpp
-    driver_api.cpp
-)
-set(C10_CUDA_HEADERS
-    CUDAAllocatorConfig.h
-    CUDACachingAllocator.h
-    CUDADeviceAssertionHost.h
-    CUDAException.h
-    CUDAFunctions.h
-    CUDAGuard.h
-    CUDAMacros.h
-    CUDAMathCompat.h
-    CUDAMiscFunctions.h
-    CUDAStream.h
-    impl/CUDAGuardImpl.h
-    impl/CUDATest.h
-)
-set(CUDA_LINK_LIBRARIES_KEYWORD PRIVATE)
-torch_cuda_based_add_library(c10_cuda ${C10_CUDA_SRCS} ${C10_CUDA_HEADERS})
-set(CUDA_LINK_LIBRARIES_KEYWORD)
-# If building shared library, set dllimport/dllexport proper.
-target_compile_options(c10_cuda PRIVATE "-DC10_CUDA_BUILD_MAIN_LIB")
-# Enable hidden visibility if compiler supports it.
-if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
-  target_compile_options(c10_cuda PRIVATE "-fvisibility=hidden")
-endif()
+  # Note: if you want to add ANY dependency to the c10 library, make sure you
+  # check with the core PyTorch developers as the dependency will be
+  # transitively passed on to all libraries dependent on PyTorch.
 
-# ---[ Dependency of c10_cuda
-target_link_libraries(c10_cuda PUBLIC c10 torch::cudart)
+  # Note: if you add a new source file/header, you will need to update
+  # torch/utils/hipify/cuda_to_hip_mappings.py for new files
+  # and headers you add
+  set(C10_CUDA_SRCS
+      CUDAAllocatorConfig.cpp
+      CUDACachingAllocator.cpp
+      CUDADeviceAssertionHost.cpp
+      CUDAException.cpp
+      CUDAFunctions.cpp
+      CUDAMallocAsyncAllocator.cpp
+      CUDAMiscFunctions.cpp
+      CUDAStream.cpp
+      impl/CUDAGuardImpl.cpp
+      impl/CUDATest.cpp
+      driver_api.cpp
+  )
```

Comment:
Perhaps a separate BE goals, but we should already have filelists for those (as we need them for bazel builds)